src/iio_wrapper.cpp

```diff
@@ -185,7 +184,8 @@ IIOWrapper::~IIOWrapper()
 
 void IIOWrapper::buffer_refill()
 {
-    iio_buffer_refill(m_device_buffer);
+    std::lock_guard<std::mutex> guard(m_mutex);
+    ssize_t ret =  iio_buffer_refill(m_device_buffer);
```

Comment:
should be added in the previous commit.
you are not doing anything with ret here... it would be a good idea to return it